### PR TITLE
Added functionality to share a message vs chat

### LINKED ISSUE #388: Allow sharing a Message vs. Chat
Our current Share feature shares the entire chat.  Let's add a way to share a single message in a chat.  Behind the scenes it can use the same code as we have now (e.g., share a chat) but only include the single message vs. everything.

Probably the simplest way to start this would be to add a new menu option to "Share message..."

Screenshot 2024-01-31 at 6 02 08 PM

### PR DESCRIPTION
**Pull request by Rachit1313**
Closes #388

### CODE REVIEW

**Comment by WangGithub0** [comment]
After I click the "share message", I only got this "message shared", but how should I share to others without a link?
image

**Comment by Rachit1313** [comment]
> After I click the "share message", I only got this "message shared", but I don't know how should I share to others without a link?
> 
> image

For now I changed it to display the URL in the prompt itself but would need to consult the team for how are we planning to display the URL for this

**Comment by kliu57** [comment]
Copy the URL to clipboard, let user know it is copied to their clipboard

**Comment by Rachit1313** [comment]
@WangGithub0 , Could you please review this whenever you get a chance?

**Comment by WangGithub0** [comment]
I wanna make sure one thing,  should shared message show the sharing time or the create time?
image

**Comment by Rachit1313** [comment]
> I wanna make sure one thing, should shared message show the sharing time or the create time? image

@humphd

**Comment by tarasglek** [comment]
This behavior is ok with me, not sure which behavior @humphd was looking for

**Comment by humphd** [comment]
I would show the created time to match what we do elsewhere.

**Comment by Rachit1313** [comment]
@kliu57 Are we good to merge this then?

**Comment by kliu57** [comment]
> @kliu57 Are we good to merge this then?

Rachit if you don't have time to fix it today, just create a follow up issue and link it to this PR. Then you can go ahead and merge it.

**Comment by kliu57** [comment]
@Rachit1313 Can you take a look at this 
image

File: src/components/Message/MessageBase.tsx [hunk 1]
```diff
@@ -123,6 +128,36 @@ function MessageBase({
...
+      });
+      copyToClipboard(url);
+    } catch (err) {
+      console.error(err);
+      error({
+        title: "Failed to Share Message",
+        message: "An error occurred while trying to share the message.",
+      });
+    }
+  }, [message.id, user, chat, info, error]);
```
Comment (WangGithub0) [inline src/components/Message/MessageBase.tsx:161]: @Rachit1313  please revise this lint and create a follow-up issue for the date, others are good for me now.

**Comment by Rachit1313** [comment]
> > @kliu57 Are we good to merge this then?
> 
> 
> 
> Rachit if you don't have time to fix the created time today, just create a follow up issue and link it to this PR. Then you can go ahead and merge it.

No I'm just confused here... so aa of now it has created time but not the shared time... Do we need shared time or created time?

**Comment by kliu57** [comment]
> > > @kliu57 Are we good to merge this then?
> > 
> > 
> > Rachit if you don't have time to fix the created time today, just create a follow up issue and link it to this PR. Then you can go ahead and merge it.
> 
> No I'm just confused here... so aa of now it has created time but not the shared time... Do we need shared time or created time(time when shared chat was created)?

We want it to be showing the created time. So both chats should show the exact same time since they are both the created time. In yumei's screenshot from yesterday afternoon you can see that the times are different. We need the times to be the same (Created time)

**Comment by kliu57** [comment]
@Rachit1313 I think there is some confusion here. So, "Created time" does not mean the time that the shared chat was created.

**Comment by kliu57** [comment]
image

**Comment by Rachit1313** [comment]
> image

Got your point... Fill fix it in a follow up PR.

**Comment by Rachit1313** [comment]
@kliu57 , We can just land it for now and create another PR for time.

**Review by WangGithub0 — APPROVED** [review]

**Comment by kliu57** [comment]
> @kliu57 , We can just land it for now and create another PR for time.

Hey rachit please file a follow up issue and then link it to this pr

**Pull request merged**